### PR TITLE
[SPARK-26109][WebUI]Duration in the task summary metrics table and the task table are different

### PR DESCRIPTION
**Pull request by shahidki31**
## What changes were proposed in this pull request?
Task summary table displays the summary of the task table in the stage page. However, the 'Duration' metrics of 'task summary' table and 'task table' are not matching. The reason is because, in the 'task summary' we display 'executorRunTime' as the duration, and in the 'task table' the actual duration of the task. Except duration metrics, all other metrics are properly displaying in the task summary.

In Spark2.2, used to show 'executorRunTime' as duration in the 'taskTable'. That is why, in summary metrics also the 'exeuctorRunTime' shows as the duration. So, we need to show 'executorRunTime' as the duration in the tasks table to follow the same behaviour as the previous versions of spark.

## How was this patch tested?
Before patch:

screenshot from 2018-11-19 04-32-06

After patch:
screenshot from 2018-11-19 04-37-39

### CODE REVIEW

**Comment by shahidki31** [comment]
retest this please

**Comment by shahidki31** [comment]
Jenkins, retest this please

**Comment by shahidki31** [comment]
retest this please

**Comment by shahidki31** [comment]
cc @vanzin @srowen Kindly review.

**Review by srowen — APPROVED** [review]
Oh that's good, I think I had noticed the inconsistency and always wondered why. I looked back even to Spark 2.0 and looks like it has been inconsistent since at least then? that is, it doesn't seem like this was an intentional change that was only partly finished. So, looks good.

**Comment by gengliangwang** [comment]
retest this please.

**Comment by shahidki31** [comment]
Seems random failure, unrelated to the PR.

File: core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala [hunk 1]
```diff
@@ -996,7 +996,7 @@ private[ui] object ApiHelper {
     HEADER_EXECUTOR -> TaskIndexNames.EXECUTOR,
     HEADER_HOST -> TaskIndexNames.HOST,
     HEADER_LAUNCH_TIME -> TaskIndexNames.LAUNCH_TIME,
-    HEADER_DURATION -> TaskIndexNames.DURATION,
+    HEADER_DURATION -> TaskIndexNames.EXEC_RUN_TIME,
```
Comment (gengliangwang) [inline core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala]: Nit: add comment to explain why it should be executor run time here

Comment (shahidki31) [reply core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala]: Thanks. Done

Comment (gengliangwang) [reply core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala]: Not big deal but there are two spaces after `SPARK-26109:`

**Review by gengliangwang — APPROVED** [review]
LGTM

File: core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala (hunk 1 above)
Comment (shahidki31) [reply core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala]: Thanks. Done

Comment (shahidki31) [reply core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala]: Done.

**Comment by shahidki31** [comment]
Thank you @gengliangwang

**Comment by shahidki31** [comment]
It is also a random failure.

**Comment by shahidki31** [comment]
again, failure unrelated to the PR.

**Comment by shahidki31** [comment]
retest this please

**Comment by srowen** [comment]
Jenkins, add to whitelist

**Comment by srowen** [comment]
Merged to master/2.4/2.3

**Comment by shahidki31** [comment]
Thank you @srowen

**Pull request closed without merging**